addons/hr_homeworking_calendar/static/src/calendar/common/calendar_common_renderer.xml

```diff
@@ -16,26 +16,32 @@
         </t>
     </t>
 
-    <div t-name="hr_homeworking_calendar.MultiCalendarButtonWorkLocation" class="o_worklocation_btn w-100 d-flex align-items-center">
-        <div class="o_homework_multi d-flex gap-1 ms-1">
+    <div t-name="hr_homeworking_calendar.MultiCalendarButtonWorkLocation" class="o_worklocation_btn w-100 d-flex align-items-center flex-grow-1">
+        <div class="d-flex gap-1 ms-1 flex-wrap">
             <t t-foreach="Object.keys(worklocation)" t-as="location" t-key="location">
-                <t t-foreach="worklocation[location]" t-as="wl" t-key="wl.id">
-                    <div class="o_homeworking_content wl_action" t-att-data-location="location" t-att-data-id="wl.id" t-att-data-employee="wl.employeeId">
-                        <i t-attf-class="o_homework_content d-flex align-items-center justify-content-center flex-wrap border border-1 rounded-circle me-0 fa {{iconMap[wl.icon] || 'fa-map-marker'}} wl_color_{{wl.colorIndex}}"/>
-                    </div>
-                </t>
-                <span class="fw-bolder" t-esc="worklocation[location][0].title"/>
+                <div class="o_homework_multi d-flex gap-1 ms-1">
+                    <t t-foreach="worklocation[location]" t-as="wl" t-key="wl.id">
+                        <div class="o_homeworking_content wl_action" t-att-data-location="location" t-att-data-id="wl.id" t-att-data-employee="wl.employeeId">
+                            <i t-attf-class="o_homework_content d-flex align-items-center justify-content-center flex-wrap border border-1 rounded-circle me-0 fa {{iconMap[wl.icon] || 'fa-map-marker'}} wl_color_{{wl.colorIndex}} wl_outline_{{wl.colorIndex}}"/>
+                        </div>
+                    </t>
+                    <span class="fw-bolder o_worklocation_span" t-esc="worklocation[location][0].title"/>
+                </div>
             </t>
         </div>
-        <button t-if="!workLocationSetForCurrentUser" class="o_worklocation_text bg-info bg-opacity-25 border-0 rounded-pill px-lg-2 py-0 text-nowrap opacity-0 opacity-100-hover wl_action" data-create="1">
-            <i class="add_wl fa fa-map-marker me-lg-1" aria-hidden="true"></i>
-        </button>
+        <div t-if="!workLocationSetForCurrentUser" class="w-100 d-flex align-items-center opacity-0 opacity-100-hover wl_action" data-create="1">
+            <button class="o_worklocation_text bg-info bg-opacity-25 border-0 rounded-pill px-lg-2 py-0 text-nowrap">
+                <i class="add_wl fa fa-map-marker me-lg-1" aria-hidden="true"></i>
+                <span t-if="Object.keys(worklocation) &lt; 1" class="d-lg-inline fw-bold location_xs_hide">Set<span class="location_xl_hide"> Location</span></span>
```

Comment:
`d-inline d-xl-none`, probably doable on the location_xs_hide as well to minimize css, I'll let you try :)